client/src/components/SideNav.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { PropTypes } from 'prop-types';
+import contactActions from '../actions/contactActions';
+import authActions from '../actions/authActions';
+import socket from '../repositories/Socket';
+import ErrorMessage from './ErrorMessage';
+
+/**
+ * SideNav component
+ */
+export class SideNav extends React.Component {
+  /**
+   * component did mount
+   * @returns {null} null
+   */
+  componentDidMount() {
+    const { getContacts, contacts } = this.props;
+    if (Object.keys(contacts).length < 1) {
+      getContacts();
+    }
+  }
+
+  /**
+   * component will receive props
+   * @param {Object} nextProps
+   * @returns {null} null
+   */
+  componentWillReceiveProps(nextProps) {
```

Comment:
componentWillReceiveProps has been deprecated in recent React versions. It goes as UNSAFE_componentWillReceiveProps() now.